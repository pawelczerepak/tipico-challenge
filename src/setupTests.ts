import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import axiosMiddleware from 'redux-axios-middleware';
import client from './apiClient';

configure({ adapter: new Adapter() });

export const mockStore = configureMockStore([thunk, axiosMiddleware(client)]);
