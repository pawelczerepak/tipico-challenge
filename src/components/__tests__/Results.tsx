import React from 'react';
import '../../setupTests';
import toJson from 'enzyme-to-json';
import ConnectedResults from '../Results';
import {MockStoreEnhanced} from 'redux-mock-store';
import {mockStore} from '../../setupTests';
import {mount} from 'enzyme';
import {Provider} from 'react-redux';

describe('Results', () => {
    let store: MockStoreEnhanced<unknown, {}>;
    beforeEach(() => {
        store = mockStore({});
    });

    it('capture snapshot', () => {
        expect(toJson(mount(<Provider store={store}><ConnectedResults/></Provider>)))
            .toMatchSnapshot();
    });
});
