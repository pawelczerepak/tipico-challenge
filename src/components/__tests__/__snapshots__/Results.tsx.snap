// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results capture snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Results)>
    <Results
      actions={
        Object {
          "searchRepositories": [Function],
          "setCurrentPage": [Function],
        }
      }
      currentPage={1}
      error={null}
      loadedPages={Array []}
      page={Array []}
      pageCount={0}
      totalCount={null}
    />
  </Connect(Results)>
</Provider>
`;
